File: backup/constantes.py
"""
Constantes centralizadas do projeto - todas em PT-BR.
"""

# =============================================================================
# APLICAÇÃO
# =============================================================================
NOME_APLICATIVO = "Calendário Kanban"
NOME_ORGANIZACAO = "DevBroCorp"
NOME_APLICACAO = "GerenciadorProV2"
ARQUIVO_BANCO_DADOS = "calendario_kanban.db"

# =============================================================================
# DIMENSÕES DA JANELA
# =============================================================================
LARGURA_JANELA = 1280
ALTURA_JANELA = 720

# =============================================================================
# ANIMAÇÕES (milissegundos)
# =============================================================================
DURACAO_FADE_IN = 400
ATRASO_FADE_IN = 50
DURACAO_FADE_OUT = 200
DURACAO_RIPPLE = 400
DURACAO_ANIMACAO_HOVER = 150

# =============================================================================
# SOMBRAS
# =============================================================================
RAIO_DESFOQUE_SOMBRA = 15
DESLOCAMENTO_Y_SOMBRA = 2
COR_SOMBRA = (0, 0, 0, 60)
DESFOQUE_SOMBRA_DIALOGO = 25
DESLOCAMENTO_Y_SOMBRA_DIALOGO = 5
COR_SOMBRA_DIALOGO = (0, 0, 0, 150)

# =============================================================================
# DIÁLOGOS
# =============================================================================
ATRASO_FADE_DIALOGO = 10
DURACAO_FADE_DIALOGO = 250
LARGURA_DIALOGO_TAREFA = 480
ALTURA_DIALOGO_TAREFA = 320
LARGURA_DIALOGO_HORARIO = 380
ALTURA_DIALOGO_HORARIO = 280

# =============================================================================
# TOAST (NOTIFICAÇÕES)
# =============================================================================
DESLOCAMENTO_X_TOAST = 20
DESLOCAMENTO_Y_TOAST = 80
DURACAO_TOAST_CURTA = 1500
DURACAO_TOAST_MEDIA = 2000

# Estilos do Toast
ESTILO_TOAST = """
    QLabel {
        background: rgba(0, 0, 0, 200);
        color: white;
        padding: 12px 24px;
        border-radius: 8px;
        font-weight: bold;
    }
"""

# =============================================================================
# HORÁRIOS
# =============================================================================
HORARIO_INICIO_PADRAO = 7
HORARIO_FIM_PADRAO = 23
HORARIO_INICIAL_SPINBOX = 9
MINUTO_INICIAL_SPINBOX = 0
FORMATO_HORA = "%H:%M"

# =============================================================================
# THREADING
# =============================================================================
MAX_THREADS_POOL = 4

# =============================================================================
# INTERFACE DO USUÁRIO
# =============================================================================
ALTURA_MINIMA_LISTA = 120
ATRASO_SOMBRA_COLUNA = 100
LARGURA_MINIMA_SPINBOX = 80
ALTURA_MINIMA_SPINBOX = 30

# =============================================================================
# FONTE
# =============================================================================
FAMILIA_FONTE = "Segoe UI"
TAMANHO_FONTE = 9

# =============================================================================
# ESTILOS CSS/QSS
# =============================================================================
ESTILO_BOTAO_ADICIONAR_HORARIO = """
    QPushButton { 
        background-color: #6c5ce7; 
        color: white; 
        border-radius: 4px; 
        padding: 6px; 
        font-weight: bold; 
    }
    QPushButton:hover { 
        background-color: #a29bfe; 
    }
"""

ESTILO_LABEL_PREVIA_HORARIO = """
    QLabel {
        background: #6c5ce7;
        color: #ffffff;
        padding: 25px; 
        border-radius: 10px; 
        font-weight: bold; 
        font-size: 36px;
        letter-spacing: 4px;
        qproperty-alignment: AlignCenter;
    }
"""

ESTILO_SPINBOX = """
    QSpinBox {
        font-size: 13px;
        font-weight: bold;
        padding: 5px;
        border: 2px solid #6c5ce7;
        border-radius: 6px;
        min-width: 70px;
        min-height: 28px;
        background-color: rgba(30,30,30,0.85);
        color: #dfe6e9;
    }
    QSpinBox:focus {
        border: 2px solid #a29bfe;
        background-color: rgba(40,40,40,0.9);
    }
    QSpinBox::up-button, QSpinBox::down-button {
        width: 18px;
        background-color: #6c5ce7;
        border: none;
    }
    QSpinBox::up-button:hover, QSpinBox::down-button:hover {
        background-color: #a29bfe;
    }
    QSpinBox::up-arrow {
        image: none;
        border-left: 4px solid transparent;
        border-right: 4px solid transparent;
        border-bottom: 4px solid white;
    }
    QSpinBox::down-arrow {
        image: none;
        border-left: 4px solid transparent;
        border-right: 4px solid transparent;
        border-top: 4px solid white;
    }
"""

ESTILO_TITULO_DIALOGO = "font-size: 16px; font-weight: bold; padding: 10px;"

ESTILO_ROTULO_STATUS = "font-weight: bold; margin-top: 10px;"

# =============================================================================
# ESTILOS DE DIÁLOGOS
# =============================================================================
ESTILO_DIALOGO_TITULO = """
    QLabel {
        font-size: 16px;
        font-weight: bold;
        color: #6c5ce7;
        padding-bottom: 5px;
    }
"""

ESTILO_DIALOGO_LABEL_INPUT = """
    QLabel {
        font-size: 14px;
        font-weight: bold;
        padding-right: 5px;
        color: #dfe6e9;
    }
"""

ESTILO_DIALOGO_BOX_PREVIA = """
    QLabel {
        font-size: 18px;
        font-weight: bold;
        color: #dfe6e9;
        border: 2px solid #6c5ce7;
        border-radius: 6px;
        background-color: rgba(30,30,30,0.85);
        padding: 10px;
        min-height: 20px;
    }
"""

# =============================================================================
# CORES DE PRIORIDADE
# =============================================================================
CORES_PRIORIDADE_VIBRANTE = {
    0: "#e74c3c",  # P0 - vermelho
    1: "#f39c12",  # P1 - laranja
    2: "#3498db",  # P2 - azul
    3: "#27ae60",  # P3 - verde
}

CORES_PRIORIDADE_SUAVE = {
    0: "#f8d7da",  # P0 - vermelho suave
    1: "#fff3cd",  # P1 - amarelo suave
    2: "#d1ecf1",  # P2 - azul claro suave
    3: "#d4edda",  # P3 - verde suave
}

NOMES_PRIORIDADE = {
    0: "Crítica",
    1: "Alta",
    2: "Média",
    3: "Baixa",
}

DESCRICOES_PRIORIDADE = {
    0: "P0 - Crítica",
    1: "P1 - Alta",
    2: "P2 - Média",
    3: "P3 - Baixa",
}

# =============================================================================
# PERIODICIDADE
# =============================================================================
NOMES_PERIODICIDADE = {
    "unica": "Única",
    "diaria": "Diária",
    "semanal": "Semanal",
    "quinzenal": "Quinzenal",
    "mensal": "Mensal",
}

LISTA_PERIODICIDADES = [
    ("unica", "Única vez"),
    ("diaria", "Diária"),
    ("semanal", "Semanal"),
    ("quinzenal", "Quinzenal"),
    ("mensal", "Mensal"),
]

INDICE_PERIODICIDADE = {
    "unica": 0,
    "diaria": 1,
    "semanal": 2,
    "quinzenal": 3,
    "mensal": 4,
}

# =============================================================================
# MENSAGENS DO SISTEMA
# =============================================================================
MSG_TITULO_OBRIGATORIO = "O título é obrigatório."
MSG_ERRO_ATUALIZAR_HORARIO = "Não foi possível atualizar o horário."
MSG_ERRO_EDITAR_HORARIO = "Erro ao editar horário."
MSG_ERRO_REMOVER_HORARIO = "Erro ao remover horário."
MSG_CONFIRMAR_EXCLUSAO_HORARIO = "Tem certeza que deseja remover o horário {horario}?\nTodos os dados dessa linha serão perdidos."
MSG_CONFIRMAR_DELETAR_TAREFA = "Deletar tarefa?"
MSG_NENHUMA_ATIVIDADE = "Nenhuma atividade agendada"
MSG_FIM_DE_SEMANA = "Fim de semana"
MSG_TAREFA_ADICIONADA = "Tarefa '{titulo}' adicionada!"
MSG_ERRO_CRIAR_TAREFA = "Erro ao criar tarefa"
MSG_TEMA_ATIVADO = "{tema} ativado"
MSG_SINCRONIZACAO_SUCESSO = "{quantidade} tarefa(s) de {dia} sincronizada(s)!"
MSG_TAREFAS_JA_SINCRONIZADAS = "Tarefas de {dia} já sincronizadas"
MSG_ERRO_SINCRONIZACAO = "Erro na sincronização: {erro}"
MSG_SEM_ATIVIDADES_FIM_SEMANA = "Fim de semana - sem atividades"
MSG_NENHUMA_ATIVIDADE_DIA = "Nenhuma atividade em {dia}"

# =============================================================================
# TOOLTIPS
# =============================================================================
TOOLTIP_TAREFA_AGENDA = "Tarefa gerada automaticamente da Agenda\nPrioridade: P{prioridade}"
TOOLTIP_TAREFA_MANUAL = "Prioridade: P{prioridade}"
TOOLTIP_ATIVIDADE = "Prioridade: P{prioridade} - {nome_prioridade}\nPeriodicidade: {periodicidade}\nCriado em: {data}"
TOOLTIP_SINCRONIZAR = "Sincroniza as atividades do dia atual para o quadro Kanban"

File: frontend/js/constantes.js
/**
 * Constantes centralizadas do projeto - todas em PT-BR.
 * Este módulo centraliza todos os valores fixos do sistema.
 */

// =============================================================================
// APLICAÇÃO
// =============================================================================
const NOME_APLICATIVO = "Calendário Kanban";
const NOME_ORGANIZACAO = "DevBroCorp";

// =============================================================================
// TEMPO E DELAYS (milissegundos)
// =============================================================================
const TEMPO_REMOCAO_TOAST = 300;
const TEMPO_ANIMACAO_SHAKE = 500;
const TEMPO_ANIMACAO_HIGHLIGHT = 1000;
const DELAY_ENTRADA_ITEM = 50;
const DURACAO_TRANSICAO_PADRAO = 300;

// =============================================================================
// PERIODICIDADE EM DIAS
// =============================================================================
const DIAS_QUINZENAL = 14;
const DIAS_SEMANA_PERIODICIDADE = 7;

// =============================================================================
// VALORES PADRÃO
// =============================================================================
const PRIORIDADE_PADRAO = 3;
const TOTAL_DIAS_SEMANA = 5;
const DEBUG_MODE = false;  // Mudar para true durante desenvolvimento

// =============================================================================
// DIMENSÕES DA JANELA
// =============================================================================
const LARGURA_JANELA = 1280;
const ALTURA_JANELA = 720;

// =============================================================================
// ANIMAÇÕES (milissegundos)
// =============================================================================
const DURACAO_FADE_IN = 400;
const ATRASO_FADE_IN = 50;
const DURACAO_FADE_OUT = 200;
const DURACAO_RIPPLE = 400;
const DURACAO_ANIMACAO_HOVER = 150;

// =============================================================================
// TOAST (NOTIFICAÇÕES)
// =============================================================================
const DURACAO_TOAST_CURTA = 1500;
const DURACAO_TOAST_MEDIA = 2000;

// =============================================================================
// HORÁRIOS
// =============================================================================
const HORARIO_INICIO_PADRAO = 7;
const HORARIO_FIM_PADRAO = 23;
const HORARIO_INICIAL_SPINBOX = 9;
const MINUTO_INICIAL_SPINBOX = 0;

// =============================================================================
// DIAS DA SEMANA
// =============================================================================
const DIAS_SEMANA = [
    "Segunda-Feira",
    "Terça-Feira",
    "Quarta-Feira",
    "Quinta-Feira",
    "Sexta-Feira"
];

const DIAS_SEMANA_CURTO = [
    "Segunda",
    "Terça",
    "Quarta",
    "Quinta",
    "Sexta"
];

// =============================================================================
// STATUS DE TAREFAS
// =============================================================================
const STATUS_TAREFA = {
    FAZER: "Fazer",
    FAZENDO: "Fazendo",
    FEITO: "Feito"
};

const STATUS_LISTA = [STATUS_TAREFA.FAZER, STATUS_TAREFA.FAZENDO, STATUS_TAREFA.FEITO];

// =============================================================================
// CORES DE PRIORIDADE
// =============================================================================
const CORES_PRIORIDADE_VIBRANTE = {
    0: "#e74c3c",  // P0 - vermelho
    1: "#f39c12",  // P1 - laranja
    2: "#3498db",  // P2 - azul
    3: "#27ae60",  // P3 - verde
};

const CORES_PRIORIDADE_SUAVE = {
    0: "#f8d7da",  // P0 - vermelho suave
    1: "#fff3cd",  // P1 - amarelo suave
    2: "#d1ecf1",  // P2 - azul claro suave
    3: "#d4edda",  // P3 - verde suave
};

const NOMES_PRIORIDADE = {
    0: "Crítica",
    1: "Alta",
    2: "Média",
    3: "Baixa",
};

const DESCRICOES_PRIORIDADE = {
    0: "P0 - Crítica",
    1: "P1 - Alta",
    2: "P2 - Média",
    3: "P3 - Baixa",
};

// =============================================================================
// PERIODICIDADE
// =============================================================================
const NOMES_PERIODICIDADE = {
    "unica": "Única",
    "diaria": "Diária",
    "semanal": "Semanal",
    "quinzenal": "Quinzenal",
    "mensal": "Mensal",
};

const LISTA_PERIODICIDADES = [
    { codigo: "unica", descricao: "Única vez" },
    { codigo: "diaria", descricao: "Diária" },
    { codigo: "semanal", descricao: "Semanal" },
    { codigo: "quinzenal", descricao: "Quinzenal" },
    { codigo: "mensal", descricao: "Mensal" },
];

// =============================================================================
// MENSAGENS DO SISTEMA
// =============================================================================
const MSG_TITULO_OBRIGATORIO = "O título é obrigatório.";
const MSG_ERRO_ATUALIZAR_HORARIO = "Não foi possível atualizar o horário.";
const MSG_ERRO_EDITAR_HORARIO = "Erro ao editar horário.";
const MSG_ERRO_REMOVER_HORARIO = "Erro ao remover horário.";
const MSG_CONFIRMAR_EXCLUSAO_HORARIO = "Tem certeza que deseja remover o horário {horario}?\nTodos os dados dessa linha serão perdidos.";
const MSG_CONFIRMAR_DELETAR_TAREFA = "Deletar tarefa?";
const MSG_NENHUMA_ATIVIDADE = "Nenhuma atividade agendada";
const MSG_FIM_DE_SEMANA = "Fim de semana";
const MSG_TAREFA_ADICIONADA = "Tarefa '{titulo}' adicionada!";
const MSG_ERRO_CRIAR_TAREFA = "Erro ao criar tarefa";
const MSG_TEMA_ATIVADO = "{tema} ativado";
const MSG_SINCRONIZACAO_SUCESSO = "{quantidade} tarefa(s) de {dia} sincronizada(s)!";
const MSG_TAREFAS_JA_SINCRONIZADAS = "Tarefas de {dia} já sincronizadas";
const MSG_ERRO_SINCRONIZACAO = "Erro na sincronização: {erro}";
const MSG_SEM_ATIVIDADES_FIM_SEMANA = "Fim de semana - sem atividades";
const MSG_NENHUMA_ATIVIDADE_DIA = "Nenhuma atividade em {dia}";

// =============================================================================
// TOOLTIPS
// =============================================================================
const TOOLTIP_TAREFA_AGENDA = "Tarefa gerada automaticamente da Agenda\nPrioridade: P{prioridade}";
const TOOLTIP_TAREFA_MANUAL = "Prioridade: P{prioridade}";
const TOOLTIP_ATIVIDADE = "Prioridade: P{prioridade} - {nome_prioridade}\nPeriodicidade: {periodicidade}\nCriado em: {data}";
const TOOLTIP_SINCRONIZAR = "Sincroniza as atividades do dia atual para o quadro Kanban";

// =============================================================================
// CHAVES LOCALSTORAGE
// =============================================================================
const STORAGE_KEYS = {
    TAREFAS: "kanban_tarefas",
    HORARIOS: "kanban_horarios",
    AGENDA: "kanban_agenda",
    TEMA: "kanban_tema_escuro",
    CONFIGURACOES: "kanban_configuracoes"
};

// =============================================================================
// EXPORTAÇÃO - Todas as constantes disponíveis globalmente
// =============================================================================
if (typeof window !== 'undefined') {
    window.Constantes = {
        // Aplicação
        NOME_APLICATIVO,
        NOME_ORGANIZACAO,
        
        // Tempos e delays
        TEMPO_REMOCAO_TOAST,
        TEMPO_ANIMACAO_SHAKE,
        TEMPO_ANIMACAO_HIGHLIGHT,
        DELAY_ENTRADA_ITEM,
        DURACAO_TRANSICAO_PADRAO,
        DURACAO_TOAST_CURTA,
        DURACAO_TOAST_MEDIA,
        DURACAO_FADE_IN,
        DURACAO_FADE_OUT,
        DURACAO_RIPPLE,
        
        // Periodicidade
        DIAS_QUINZENAL,
        DIAS_SEMANA_PERIODICIDADE,
        
        // Dias e Status
        DIAS_SEMANA,
        DIAS_SEMANA_CURTO,
        STATUS_TAREFA,
        STATUS_LISTA,
        TOTAL_DIAS_SEMANA,
        
        // Prioridades
        PRIORIDADE_PADRAO,
        CORES_PRIORIDADE_VIBRANTE,
        CORES_PRIORIDADE_SUAVE,
        NOMES_PRIORIDADE,
        DESCRICOES_PRIORIDADE,
        
        // Periodicidade
        NOMES_PERIODICIDADE,
        LISTA_PERIODICIDADES,
        
        // Armazenamento
        STORAGE_KEYS,
        
        // Mensagens
        MSG_TITULO_OBRIGATORIO,
        MSG_CONFIRMAR_DELETAR_TAREFA,
        MSG_CONFIRMAR_EXCLUSAO_HORARIO,
        MSG_NENHUMA_ATIVIDADE,
        MSG_FIM_DE_SEMANA,
        MSG_SEM_ATIVIDADES_FIM_SEMANA
    };
}
